app/src/main/java/org/xbmc/kore/utils/PluginUrlUtils.java

```diff
@@ -41,29 +41,9 @@ public class PluginUrlUtils {
     public static String toDefaultYouTubePluginUrl(Uri playUri) {
         String host = playUri.getHost();
 
-        if (host.endsWith("youtube.com")) {
-            String videoId = playUri.getQueryParameter("v");
-            String playlistId = playUri.getQueryParameter("list");
-            Uri.Builder pluginUri = new Uri.Builder()
-                    .scheme("plugin")
-                    .authority("plugin.video.youtube")
-                    .path("play/");
-            boolean valid = false;
-            if (videoId != null) {
-                valid = true;
-                pluginUri.appendQueryParameter("video_id", videoId);
-            }
-            if (playlistId != null) {
-                valid = true;
-                pluginUri.appendQueryParameter("playlist_id", playlistId)
-                        .appendQueryParameter("order", "default");
-            }
-            if (valid) {
-                return pluginUri.build().toString();
-            }
-        } else if (host.endsWith("youtu.be")) {
-            return "plugin://plugin.video.youtube/play/?video_id="
-                   + playUri.getLastPathSegment();
+        if (host.endsWith("youtube.com") || host.endsWith("youtu.be")) {
+            return "plugin://plugin.video.youtube/uri2addon/?uri="
+                   + playUri.toString();
```

Comment:
Don't you need to percent-encode it first?

(Sorry for the previous comment, clicked on the wrong place)

Reply:
Hi @peat-psuwit ,
I am using a patched version since I opened this PR and it works without any issues like this.
I also captured the communication and I don't see any issues there either:
```
03:39:00.924342 IP 192.168.1.104.47800 > 192.168.1.145.9090: Flags [P.], seq 60:221, ack 90, win 128, options [nop,nop,TS val 3627711265 ecr 166447552], length 161
E.../,@.@......h......#.3...9..M....o......
.:{!	...{"jsonrpc":"2.0","method":"Player.Open","id":2,"params":{"item":{"file":"plugin://plugin.video.youtube/uri2addon/?uri=https://youtube.com/watch?v=qDWkpDKA3eQ"}}}
03:39:00.967090 ethertype IPv4, IP 192.168.1.145.9090 > 192.168.1.104.47800: Flags [P.], seq 90:128, ack 221, win 235, options [nop,nop,TS val 166447565 ecr 3627711265], length 38
E..Z.!@.@..3.......h#...9..M3..<...........
	....:{!{"id":2,"jsonrpc":"2.0","result":"OK"}
```
@MoojMidge, can you confirm the url encoding is strictly necessary?

Reply:
> @MoojMidge, can you confirm the url encoding is strictly necessary?

For every type of url that the plugin supports? No.
For any type of url that the plugin supports? Yes.

As a simply example try `https://youtube.com/watch?v=qDWkpDKA3eQ&t=1m6s` and see the difference that url encoding makes.

Reply:
I've pushed a new version, the resulting request looks like this:
```
11:59:08.012913 IP realme-9.byt.danman.eu.39418 > 192.168.1.145.9090: Flags [P.], seq 138:312, ack 77, win 128, options [nop,nop,TS val 3635122324 ecr 173949325], length 174
E...IY@.@.ls...h......#.......o.....-......
....
^A.{"jsonrpc":"2.0","method":"Player.Open","id":29,"params":{"item":{"file":"plugin://plugin.video.youtube/uri2addon/?uri=https%3A%2F%2Fyoutube.com%2Fwatch%3Fv%3DCExTREBHktI"}}}

```